src/Rules/CallableExistsCheck.php

```diff
@@ -0,0 +1,124 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules;
+
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Name;
+use PhpParser\Node\Scalar\MagicConst\Class_;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Broker\Broker;
+use PHPStan\Type\ObjectType;
+
+class CallableExistsCheck
+{
+
+	/**
+	 * @var \PHPStan\Broker\Broker
+	 */
+	private $broker;
+
+	public function __construct(Broker $broker)
+	{
+		$this->broker = $broker;
+	}
+
+	/**
+	 * @param \PhpParser\Node\Arg $argument
+	 * @param \PHPStan\Analyser\Scope $scope
+	 * @param string $msgPrefix
+	 * @return string[]
+	 */
+	public function checkCallableArgument(
+		Arg $argument,
+		Scope $scope,
+		string $msgPrefix
+	): array
+	{
+		$targetType = null;
+		$targetValue = null;
+		$targetMethod = null;
+		$targetIsStaticMethod = false;
+		if ($argument->value instanceof Array_) {
+			$arrayItemsCount = count($argument->value->items);
+			if ($arrayItemsCount < 2) {
+				return [$msgPrefix . "array doesn't have required 2 items."];
+			}
+			if ($arrayItemsCount > 2) {
+				return [$msgPrefix . 'array has too much items, only two items expected.'];
+			}
+			$item1 = $argument->value->items[0]->value;
+			$item2 = $argument->value->items[1]->value;
+
+			if ($item1 instanceof String_) {
+				$targetType = new ObjectType($item1->value);
+				$targetValue = $item1->value;
+				$targetIsStaticMethod = true;
+			} elseif ($item1 instanceof Variable) {
```

Comment:
Does not have to be variable, e.g. `[$this->object, 'method']`. This should be the fallback `else` branch.